sqlglot/lineage.py

```diff
@@ -276,8 +276,81 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next(
+                (p for p in scope.pivots if p.alias_or_name == c.table and not p.unpivot), None
+            )
+            if pivot and isinstance(pivot.args.get("columns"), list):
+                # The source is a pivot operation, so we need to trace back to the aggregated column
+                pivot_column_mapping = {}
+                pivot_aggs_count = len(pivot.expressions)
+                columns = pivot.args["columns"]
+                columns_count = len(columns)
+
+                if c.alias_or_name not in [i.this for i in columns]:
+                    # The column is not in the pivot, so it must be an implicit column of the pivoted source.
+                    table = pivot.parent.name
```

Comment:
Probably safer to do `table = pivot.parent.alias_or_name` here to account for pivoted derived tables as well. We already do this in L330.

Reply:
Changed it